cluster/nginx.go

```diff
@@ -167,35 +197,18 @@ func DeployNginxResolver(shouldUpdate bool) error {
 	if err != nil {
 		return err
 	}
-	if shouldUpdate {
-		//Updating the version of the nginx-resolver
-		fmt.Println("creating nginx-resolver deployment with updated rules")
-		// nginxResolverOldVersion := nginxResolverVersion
-		nginxResolverVersion = fmt.Sprintf(`%s-%s`, nginxLBName, strings.ToLower(RandomCharString(6)))
-		nginxLBDeployment.Spec.Template.ObjectMeta.Labels["app"] = nginxResolverVersion
-		nginxLBDeployment.ObjectMeta.Name = nginxResolverVersion
-		if _, err = extV1beta1API(clientset).Deployments("default").Create(&nginxLBDeployment); err != nil {
-			return err
-		}
-		//Update nginx-resolver service to route traffic to the new nginx pods
-		nginxService, err := clientset.CoreV1().Services("default").Get("nginx-resolver", metav1.GetOptions{})
-		if err != nil {
-			return err
-		}
-		nginxService.Spec.Selector["app"] = nginxResolverVersion
-		clientset.CoreV1().Services("default").Update(nginxService)
-		// Delete nginx-resolver deployment and wait until all its pods
-		// are deleted too. This is to ensure that the creation of the
-		// deployment results in new set of pods that have read the
-		// updated rules
-		waitCh := DeleteNginxLBDeployments(clientset, nginxResolverVersion)
-		// waiting for the delete of the nginx-resolver deployment to complete
-		<-waitCh
-	} else {
-		if _, err = extV1beta1API(clientset).Deployments("default").Create(&nginxLBDeployment); err != nil {
-			return err
-		}
-	}
+	nginxResolverVersion = fmt.Sprintf(`%s-%s`, nginxLBName, strings.ToLower(RandomCharString(6)))
+	waitCreateCh := CreateNginxResolverDeployment(clientset, nginxResolverVersion)
+	<-waitCreateCh
+	waitUpdateCh := UpdateNginxResolverService(clientset)
+	<-waitUpdateCh
+	// Delete nginx-resolver deployment and wait until all its pods
```

Comment:
We can relax this since the service is updated to point to the new nginx deployment. Deleting the deployment where k8s eventually delets the pods that belonging to it allows us to avoid waiting for the pods to be deleted (in the foreground).